lib/caffeinate/dripper/drip.rb

```diff
@@ -47,7 +47,34 @@ def drips
         #
         # @option options [Symbol] :using Set to `:parameters` if the mailer action uses ActionMailer::Parameters
         def drip(action_name, options = {}, &block)
-          drip_collection.register(action_name, options, &block)
+          drip_collection.register(action_name, options, ::Caffeinate::Drip, &block)
+        end
+
+        # Register a Periodical drip on the Dripper
+        #
+        #   periodical :pay_your_invoice, every: 1.day, start: 0.hours, if: :invoice_unpaid?
+        #
+        # @param action_name [Symbol] the name of the mailer action
+        # @param [Hash] options the options to create a drip with
+        # @option options [String] :mailer_class The mailer_class
+        # @option options [Symbol|Proc|ActiveSupport::Duration] :every How often the mailing should be created
+        # @option options [Symbol|Proc] :if If the periodical should create another mailing
+        # @option options [Symbol|Proc] :start The offset time to start the clock (only used on the first mailing creation)
```

Comment:
I like this API 👍 

- `:every` a static value or proc that yields an _amount of time_ that ought to be used for the 'next' message in the sequence
  - I wonder if this should also support using a specific time or date rather than an _amount of time_ — we can probably just call `.respond_to? :from_now` to see if it's a specific date or just an `ActiveSupport::Duration`
- `:if` a proc that, if yields false, ends the periodical altogether — probably should run before the actual send so that it doesn't send the message at hand if false
- `:start` a static value or proc that yields an _amount of time_ that will be added to 'right now' to set the firing date for the _first_ message _only_.
  - Same note here to consider adding support for specific times/dates as well as durations

I wonder if we should add the `until` logic to this PR?

- `:until` a proc that, if yields true, ends the periodical altogether (probably also before_drip)
  - Thinking that true == end because the usage will probably be like `until: -> { subscriber.is_opted_out? }` or `until: -> { Date.current.year > 2023 }`

Reply:
re: `deliver_class` — yeah, something like that.